modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/egr/Tile.java

```diff
@@ -0,0 +1,466 @@
+    }
+
+
+    private WritableRaster buildSolidRaster(int tileWidth, int tileHeight, int value)
+    {
+        SampleModel sampleModel = getMPSampleModel(tileWidth, tileHeight);
+
+        // build the raster
+        WritableRaster newRaster = RasterFactory.createWritableRaster(
+                sampleModel, new java.awt.Point(0, 0));
+
+        // sanity checks
+        int dataType = sampleModel.getTransferType();
+        int numBands = sampleModel.getNumBands();
+        if (dataType != DataBuffer.TYPE_BYTE)
+        {
+            throw new IllegalArgumentException("The code works only if the sample model data type is BYTE");
+        }
+        if (numBands != 1)
+        {
+            throw new IllegalArgumentException("The code works only for single band rasters!");
+        }
+
+        if (value != 0)
+        {
+            // flood fill
+            int w = sampleModel.getWidth();
+            int h = sampleModel.getHeight();
+            int[] data = new int[w * h];
+
+            Arrays.fill(data, value);
+            newRaster.setSamples(0, 0, w, h, 0, data);
+        }
+
+        return newRaster;
+    }
+
+    private SampleModel getMPSampleModel(int tileWidth, int tileHeight)
+    {
+        SampleModel sampleModel;
+
+        if ((tileWidth == ROIExcessGranuleRemover.DEFAULT_TILE_SIZE) && (tileHeight == ROIExcessGranuleRemover.DEFAULT_TILE_SIZE))
+        {
+            sampleModel = DEFAULT_PACKED_SAMPLE_MODEL;
+        }
+        else
+        {
+            String key = tileWidth + "x" + tileHeight;
+            sampleModel = mpSampleModelCache.get(key);
+            if (sampleModel == null)
+            {
+                sampleModel = new MultiPixelPackedSampleModel(DataBuffer.TYPE_BYTE, tileWidth, tileHeight, 1);
+                mpSampleModelCache.put(key, (MultiPixelPackedSampleModel) sampleModel);
+            }
+        }
+
+        return sampleModel;
+    }
+
+    public void draw(Shape projectedShape)
+    {
+        draw(projectedShape, false);
+    }
+
+    public void draw(Shape projectedShape, boolean inverted)
+    {
+        initGraphics(inverted);
+
+        Color drawColor = inverted ? Color.BLACK : Color.WHITE;
+        graphics.setColor(drawColor);
+        graphics.fill(projectedShape);
+    }
+    
+    public void draw(/* another Raster here I suppose? Maybe a tile, maybe mis-aligned with this one */)
+    {
+        initRaster(false);
+        
+        // flip bits here, if the tile is aligned we can do int math, otherwise bit by bit...
+    }
+
+    public void dispose()
+    {
+        if (graphics != null)
+        {
+            graphics.dispose();
+        }
+        bi = null;
+        raster = null;
+    }
+
+    @Override
+    public String toString()
+    {
+        return "Tile[" + col + 'x' + row + ':' + coverageCount + ']';
+    }
+
+    WritableRaster getRaster()
+    {
+        return raster;
+    }
+    
+    public Rectangle getTileArea() {
+        return tileArea;
+    }
+
+}
```

Comment:
Not an outstading issue but the class has been formatted using a different indentation style/formatting template where "{" lives on a separate line after method signature and if/else checks.